### DISCUSSION TITLE
Add Server Mobile Data Connection Support

### DISCUSSION THREAD

**Discussion by rigunie** [UmerCodez/SensorServer#55] [Ideas]
I was trying to use sensor server while connected via mobile data but I think it only works with a Wi-Fi connection.

It will be great if the developers add a version capable of connect to the server over mobile data. Is there any limitation or know issue that blocks this implementation?

Thanks for your attention

**Comment by UmerCodez** [comment]
Yes, there is a limitation, see https://stackoverflow.com/a/25970755

**Reply by rigunie** [reply]
Thanks for your answer. Indeed there is this limitation, but I think this only apply when using different subnet. 

In my case I intend to connect two devices in the same subnet over mobile data.

When I try to start the server while connected with mobile data, there is an error message that says: "Please Enable Wi-Fi". And the server doesn't start. 

Does the limitation described apply in that case?

**Reply by UmerCodez** [reply]
Apologies for the delayed response. This application is specifically designed for sensor streaming within a local Wi-Fi network. If using Wi-Fi isn't feasible in your situation, the app fortunately offers a hotspot option which will allow server to listen on the IP address assigned to the hotspot, allowing other devices to connect and access sensor data.
If your goal is to access sensor data remotely from over the internet, you'll need to configure port forwarding on your Wi-Fi router. This involves directing incoming traffic from a specific port on the internet to the device running the server app within your local network. However, setting up port forwarding requires administrative access to your Wi-Fi router.

Directly pinging devices on cellular networks can be challenging due to NAT implemented by cellular carriers. This hides individual device IP addresses behind a shared network address, making direct communication more complex. Have you tried pinging one device from other over cellular network?
